app/authentication/jwt_decoder.py

```diff
@@ -1,179 +0,0 @@
-import json
-
-import jwt
-from cryptography.exceptions import InternalError
-from cryptography.exceptions import InvalidTag
-from cryptography.hazmat.backends.openssl.backend import backend
-from cryptography.hazmat.primitives import serialization
-from structlog import get_logger
-
-from app.authentication.invalid_token_exception import InvalidTokenException
-from app.authentication.no_token_exception import NoTokenException
-from app.cryptography.jwe_decryption import JWERSAOAEPDecryptor
-from app.utilities import strings
-
-IV_EXPECTED_LENGTH = 12
-CEK_EXPECT_LENGTH = 32
-
-logger = get_logger()
-
-
-class JWTDecryptor(JWERSAOAEPDecryptor):
-    """
-    JWT signed with JWS RS256 And encrypted with JWE RSA-OAEP
-    """
-    def __init__(self, private_key, private_key_password, public_key):
-        # pylint: disable=maybe-no-member
-        # password and key variables are dynamically assigned
-        if private_key and private_key_password and public_key:
-
-            super().__init__(private_key, private_key_password)
-            # oddly the python cryptography library needs these as bytes string
-            rrm_public_key_as_bytes = strings.to_bytes(public_key)
-            self.rrm_public_key = serialization.load_pem_public_key(
-                rrm_public_key_as_bytes,
-                backend=backend,
-            )
-        else:
-            raise OSError('keymat not configured correctly')
-
-    def decode_signed_jwt_token(self, signed_token, leeway):
-        try:
-            if signed_token:
-                logger.debug("decoding signed jwt", jwt_token=strings.to_str(signed_token))
-                self._check_token(signed_token)
-                token = jwt.decode(signed_token, self.rrm_public_key, algorithms=['RS256'],
-                                   leeway=leeway)
-                if not token:
-                    raise InvalidTokenException("Missing Payload")
-                return token
-            else:
-                raise NoTokenException("JWT Missing")
-        except (jwt.DecodeError,
-                jwt.exceptions.InvalidAlgorithmError,
-                jwt.exceptions.ExpiredSignatureError,
-                jwt.exceptions.InvalidIssuedAtError) as e:
-            raise InvalidTokenException(repr(e))
-
-    def _check_token(self, token):
-        token_as_str = strings.to_str(token)
-        if token_as_str.count(".") != 2:
-            raise InvalidTokenException("Invalid Token")
-
-        # header_data, payload_data, signature_data
-        header_data, payload_data, _ = token_as_str.split('.', maxsplit=2)
-
-        self._check_headers(header_data)
-        self._check_header_values(token_as_str)
-        self._check_payload(payload_data)
-
-    @staticmethod
-    def _check_header_values(token):
-        header = jwt.get_unverified_header(token)
-
-        if not header:
```

Comment:
Yes the library check all of this for us. the tests are still present that used to test this.